src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/PacketHandle.Windows.cs

```diff
@@ -22,21 +20,33 @@ internal readonly ref struct PacketHandle
     {
         public const int NativePointerType = 1;
         public const int NativePacketType = 2;
+#if NET
```

Comment:
So, I saw a similar scenario in the SNIHandle class, and I'm not sure this is a good approach. My thoughts here is that this is a good example where we should have an abstract class and a ManagedPacketHandle and a NativePacketHandle. These are very lightweight classes that just wrap the actual packet handle, which makes me wonder if they're even necessary in the first place. So, my big question is: 
* option a, should we just merge it like this and maybe address it later
* option b, should we introduce some abstraction
* option c, should we consider removing the handle class altogether

Reply:
I think we can set option b aside - this is a ref struct, so it can't inherit anything. There might be a path to replace this via generics, but there's already a design gap between netcore and netfx. I think we should merge the type as-is, finish the merge of TdsParserStateObject and TdsParserStateObjectNative, then revisit it.

Reply:
Ok, option A it is
I keep forgetting about ref structs ... kinda annoying technicalities imho